### PR TITLE
Implement History Edit and Delete functionality

### PR DESCRIPTION
**Pull request by masanori-satake**
This change implements the requested history editing and deletion features for QuickLog-Solo.

Key implementations:
1. **Interactive History**: Log items (except active tasks) are now clickable, opening a Material 3-styled dialog.
2. **Temporal Consistency**:
   - Editing a log's start time automatically updates the end time of a contiguous previous log, preventing gaps.
   - Deleting a log shifts the start time of the next log to the deleted log's original start point, maintaining continuity.
   - Intentional gaps between logs (e.g., manually stopped periods) are respected and preserved.
3. **Minute-Level Precision**: While the UI operates in minutes for better UX, the internal logic ensures that seconds are synchronized (typically zeroed out) to avoid sub-second fragments.
4. **Validation**: The UI prevents out-of-range edits (outside the current day or crossing over neighboring logs' boundaries) and displays a warning message.
5. **Localization**: Full support for Japanese and English, including specific labels for "Stop" markers.
6. **Robustness**: Core logic was extracted to `shared/js/logic.js` and verified with comprehensive unit tests covering propagation and deletion edge cases.

---
*PR created automatically by Jules for task [13209328610680474723](https://jules.google.com/task/13209328610680474723) started by @masanori-satake*

### CODE REVIEW

File: projects/app/js/app.js [hunk 1]
```diff
@@ -191,6 +191,105 @@ async function endTask() {
...
+            isValid = false;
+        }
+        // 3. Must be before current end (if not stop marker)
+        if (!isStopMarker && newTs > log.endTime) {
+            isValid = false;
+        }
+        // Special case for stop marker: must not exceed previous start if contiguous
+        if (isStopMarker && prevLog && prevLog.endTime === log.startTime && newTs < prevLog.startTime) {
+            isValid = false;
+        }
```
[the remark this answers is not in the document]
Comment (masanori-satake) [reply projects/app/js/app.js:255]: @jules
PRコメントを参考に改善してください。

File: shared/js/logic.js [hunk 2]
```diff
@@ -425,3 +425,54 @@ export async function pauseTaskLogic(activeTask) {
...
+    const nextLog = currentIndex < sortedLogs.length - 1 ? sortedLogs[currentIndex + 1] : null;
+
+    const startTs = log.startTime;
+    await dbDelete(STORE_LOGS, logId);
+
+    // Propagation for deletion: update next item's start time
+    if (nextLog) {
+        nextLog.startTime = startTs;
+        await dbPut(STORE_LOGS, nextLog);
+    }
```
[the remark this answers is not in the document]
Comment (masanori-satake) [reply shared/js/logic.js]: @jules
PRコメントを参考に改善してください。
